src/utils.jl

```diff
@@ -742,3 +742,116 @@ modules(m) = [x for x in Functors.fcollect(m) if !isleaflike(x)]
 isleaflike(x) = Functors.isleaf(x)
 isleaflike(::Tuple{Vararg{<:Number}}) = true
 isleaflike(::Tuple{Vararg{<:AbstractArray{<:Number}}}) = true
+
+"""
+    patience(predicate, patience)
+
+Return a function that internally counts by one when
+`predicate(...) == true`, otherwise the count is reset to zero.
+If the count is greater than or equal to `patience`,
+the function returns `true`, otherwise it returns `false`.
+
+# Examples
+```jldoctest
+julia> loss() = rand();
+
+julia> trigger = Flux.patience(() -> loss() < 1, 3);
+
+julia> Flux.@epochs 10 begin
+       trigger() && break
+       end
+[ Info: Epoch 1
+[ Info: Epoch 2
+[ Info: Epoch 3
+```
+"""
+function patience(predicate, patience)
+  on_trigger = let count = 0
+    (args...; kwargs...) -> begin
+      count = predicate(args...; kwargs...) ? count + 1 : 0
+
+      return count >= patience
+    end
+  end
+end
```

Comment:
Just so that the argument and function name aren't the same.

```suggestion
function patience(predicate, wait)
  on_trigger = let count = 0
    (args...; kwargs...) -> begin
      count = predicate(args...; kwargs...) ? count + 1 : 0

      return count >= wait
    end
  end
end
```